stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/listener/ClientStreamingListener.java

```diff
@@ -48,38 +48,10 @@ public ClientStreamingListener(Descriptors.MethodDescriptor methodDescriptor, Ma
     @Override
     public StreamObserver<Message> invoke(StreamObserver<Message> responseObserver) {
         Resource onOpen = resourceMap.get(GrpcConstants.ON_OPEN_RESOURCE);
-        CallableUnitCallback callback = new GrpcCallableUnitCallBack(responseObserver, Boolean.FALSE);
+        CallableUnitCallback callback = new GrpcCallableUnitCallBack(null);
         Executor.submit(onOpen, callback, null, null, computeMessageParams
                 (onOpen, null, responseObserver));
         
-        return new StreamObserver<Message>() {
-            @Override
-            public void onNext(Message value) {
-                Resource onMessage = resourceMap.get(GrpcConstants.ON_MESSAGE_RESOURCE);
-                CallableUnitCallback callback = new GrpcCallableUnitCallBack(responseObserver, Boolean.FALSE);
-                Executor.submit(onMessage, callback, null, null, computeMessageParams
-                        (onMessage, value, responseObserver));
-            }
-            
-            @Override
-            public void onError(Throwable t) {
-                Resource onError = resourceMap.get(GrpcConstants.ON_ERROR_RESOURCE);
-                onErrorInvoke(onError, responseObserver, t);
-            }
-            
-            @Override
-            public void onCompleted() {
-                Resource onCompleted = resourceMap.get(GrpcConstants.ON_COMPLETE_RESOURCE);
-                if (onCompleted == null) {
-                    String message = "Error in listener service definition. onError resource does not exists";
-                    LOG.error(message);
-                    throw new RuntimeException(message);
-                }
-                
-                CallableUnitCallback callback = new GrpcCallableUnitCallBack(responseObserver, isEmptyResponse());
-                Executor.submit(onCompleted, callback, null, null, computeMessageParams
-                        (onCompleted, null, responseObserver));
-            }
-        };
+        return new DefaultStreamObserver(resourceMap, responseObserver);
```

Comment:
Shall we remove unused Logger from this class?

Reply:
Fixed with commit 30f5bc77a909cf82f701ac2766b84b76b2f4c0a1
Thanks